From: Alessandro Bono <alessandro.bono369@gmail.com>
Date: Fri, 20 Oct 2023 22:00:40 +0200
Subject: animationUtils: Allow marking animations as required

There are cases when we want to mark an animation as required. For
example, we want the "Locate Pointer" animation to work even when
the animation is marked as disabled. Take this into account when
adjusting the animation time.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2986>
(cherry picked from commit 27878aa0965f2dfc503bac4e0a12a8a2fac3e548)
(cherry picked from commit a5513de146f8ed5bf1dc003771169fb34390da5b)

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/2039801
---
 js/misc/animationUtils.js | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/js/misc/animationUtils.js b/js/misc/animationUtils.js
index a7a0ea8..cf68896 100644
--- a/js/misc/animationUtils.js
+++ b/js/misc/animationUtils.js
@@ -13,14 +13,20 @@ const N_WIGGLES = 3;
  * adjustAnimationTime:
  *
  * @param {number} msecs - time in milliseconds
+ * @param {object} params - optional parameters
+ * @param {boolean=} params.animationRequired - whether to ignore the enable-animations setting
  *
  * Adjust `msecs` to account for St's enable-animations
  * and slow-down-factor settings
  */
-export function adjustAnimationTime(msecs) {
+export function adjustAnimationTime(msecs, params) {
+    params = Params.parse(params, {
+        animationRequired: false,
+    });
+
     const settings = St.Settings.get();
 
-    if (!settings.enable_animations)
+    if (!settings.enable_animations && !params.animationRequired)
         return 0;
     return settings.slow_down_factor * msecs;
 }
